.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,115 @@
+      OIDC_HRI_INTERNAL_CLIENT_ID: e1284680-4edf-44d2-b65c-d63097906fae
+      OUTPUT_TOPIC: ingest.test.flink-validation-passthrough.out
+      NOTIFICATION_TOPIC: ingest.test.flink-validation-passthrough.notification
+
+      ACCESS_TOKEN_USER: ${{ secrets.ACCESS_TOKEN_USER }}
+      ACCESS_TOKEN_VALUE: ${{ secrets.ACCESS_TOKEN_VALUE }}
+      CLOUD_API_KEY: ${{ secrets.CLOUD_API_KEY }}
+      ELASTIC_PASSWORD: ${{ secrets.ELASTIC_PASSWORD }}
+      ELASTIC_USER: ${{ secrets.ELASTIC_USER }}
+      OIDC_HRI_DATA_INTEGRATOR_CLIENT_SECRET: ${{ secrets.OIDC_HRI_DATA_INTEGRATOR_CLIENT_SECRET }}
+      OIDC_HRI_INTERNAL_CLIENT_SECRET: ${{ secrets.OIDC_HRI_INTERNAL_CLIENT_SECRET }}
+      SASL_PLAIN_PASSWORD: ${{ secrets.SASL_PLAIN_PASSWORD }}
+
+    steps:
+      - name: Set Environment Variables
+        run: |
+          echo "FLINK_URL=https://flink-$NAMESPACE.wh-hri.dev.watson-health.ibm.com/v1" >> $GITHUB_ENV
+          echo "HRI_INGRESS_URL=https://mgmt-api-$NAMESPACE.wh-hri.dev.watson-health.ibm.com/hri" >> $GITHUB_ENV
+
+      - name: Set Branch Name
+        uses: nelonoel/branch-name@v1.0.1
+
+      - name: Install Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt-hotspot'
+          java-version: '8'
+
+      - name: Install Ruby 2.6.5
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6.5
+          bundler-cache: false
+
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Copyright Check
+        run: ./copyrightCheck.sh
+
+      - name: Gradle Build
+        run: |
+          chmod +x ./gradlew
+          ./gradlew clean build shadowJar --info
+
+      - name: Run Smoke Tests
+        run: |
+          chmod +x ./run-smoketests.sh
+          ./run-smoketests.sh
+
+      - name: Install Ruby Gems, Run IVT
+        run: |
+          gem install bundler
+          BUNDLE_GEMFILE="./test/Gemfile" bundle install
+          gem specific_install -l https://github.com/Alvearie/hri-test-helpers.git main
+          curl -sL https://ibm.biz/idt-installer | bash
+          ibmcloud login --apikey $CLOUD_API_KEY -r us-east || { echo 'IBM Cloud CLI login failed!'; exit 1; }
+          ibmcloud plugin install event-streams
+          ibmcloud es init -i ${ES_INSTANCE}
+          ./run-ivttests.sh
+
+      - name: Run Nightly Tests
+        if: github.ref == 'refs/heads/develop'
+        run: |
+          ibmcloud ks cluster config --cluster hri-dev1-cluster-1
+          ./gradlew copyNightlyTestDependencies
+          ./run-nightlytests.sh
+
+      - name: Upload Test Results
+        if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop'
+        run: |
+          ./test/spec/upload_test_reports.rb IVT
+          ./test/spec/upload_test_reports.rb Nightly
+
+      - name: Gradle Publish
+        if: ${{ github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/') }}
```

Comment:
Should we also publish the artifiacts on the `main` branch?

If not, why not?

Reply:
I think publishing artifacts on the main branch is what we'd (essentially) do for a release, but we'd do it with a tag instead of on pushes to the main branch. I tried to mimic the old travis build's logic, but that's my understanding of why the condition would be like that.

Reply:
OK. No Prob.